nodes/src/nodes/db_neo4j/IInstance.py

```
@@ -33,42 +15,152 @@
+        question = args.get('question')
+        if not question or not isinstance(question, str) or not question.strip():
+            raise ValueError('"question" is required and must be a non-empty string')
+
+        limit = _clamp_limit(args.get('limit'))
+        result = self._buildCypherQuery(question.strip(), limit=limit)
+        is_valid = _parse_is_valid(result.get('isValid', False))
+        cypher = result.get('query', '')
+
+        if is_valid and cypher and _is_cypher_safe(cypher):
+            return {'cypher': cypher, 'valid': True}
+        elif is_valid and cypher:
```

Ruff (0.15.7)

[warning] Unnecessary `elif` after `return` statement

Remove unnecessary `elif`

(RET505)